app/components/MultiSelectMap.tsx

```diff
@@ -0,0 +1,224 @@
+      return;
+    }
+
+    const map = new L.Map(containerRef.current, {
+      center: L.latLng(position),
+      zoom: 13,
+      layers: [
+        L.tileLayer("https://{s}.data.amsterdam.nl/topo_rd/{z}/{x}/{y}.png", {
+          attribution: "",
+          subdomains: ["t1", "t2", "t3", "t4"],
+          tms: true,
+        }),
+      ],
+      zoomControl: false,
+      maxZoom: 16,
+      minZoom: 6,
+      crs: getCrsRd(),
+      maxBounds: [
+        [52.25168, 4.64034],
+        [52.50536, 5.10737],
+      ],
+    });
+
+    // Remove Leaflet link from the map
+    map.attributionControl.setPrefix(false);
+
+    createdMapInstance.current = true;
+    setMapInstance(map);
+
+    map.on("moveend", () => {
+      // setDisplayAlert(true);
+      setPosition([map.getCenter().lat, map.getCenter().lng]);
+    });
+
+    // On component unmount, destroy the map and all related events
+    return () => {
+      if (mapInstance) mapInstance.remove();
+    };
+  }, []);
+
+  // Add the markers
+  useEffect(() => {
+    if (mapInstance === null) {
+      return;
+    }
+
+    const loadData = async () => {
+      const zoom = mapInstance.getZoom();
+
+      // Only query for parkeervakken if map is zoomed in to avoid large query results
+      if (zoom >= 12) {
+        const bounds = mapInstance.getBounds();
+        const url = formatWfsUrl(bounds);
+        try {
+          console.log(url);
+          const res = await fetch(url);
+          console.log(res);
+          const geo = await res.json();
+          console.log("b");
+          setMarkerData((prev) => {
+            const fetched = geo.features;
+            const fetchedIds = new Set(fetched.map((f) => f.properties.id));
+
+            // Include markers that are selected but not in the current query result
+            const preservedSelected = prev.filter(
+              (f) =>
+                selectedMarkers.includes(f.properties.id) &&
+                !fetchedIds.has(f.properties.id)
+            );
+            return [...fetched, ...preservedSelected];
+          });
+        } catch (e) {
+          console.error("Failed ot fetch parking data: ", e);
+        }
+      } else {
+        // Delete all markers except for the ones already selected
+        setMarkerData((prev) =>
+          prev.filter((f) => selectedMarkers.includes(f.properties.id))
+        );
+      }
+    };
+
+    loadData();
+
+    const handleMoveEnd = () => {
+      setPosition([mapInstance.getCenter().lat, mapInstance.getCenter().lng]);
+      loadData();
+    };
+
+    mapInstance.on("moveend", handleMoveEnd);
+
+    return () => {
+      mapInstance.off("moveend", handleMoveEnd);
+    };
+  }, [mapInstance, selectedMarkers]);
+
+  // Add the markers
+  useEffect(() => {
+    if (mapInstance === null) {
+      return;
+    }
+
+    const layerGroup = L.geoJson(markerData, {
+      style: {
+        opacity: 0.5,
+      },
+      onEachFeature: function (_feature, layer) {
```

Comment:
types should be: onEachFeature: function (_feature: GeoJSONFeature, layer: L.Polygon) {